packages/next-rest-framework/src/generate/utils.ts

```diff
@@ -36,10 +38,20 @@ export const compileEndpoints = async () => {
     format: 'cjs',
     platform: 'node',
     outdir: NEXT_REST_FRAMEWORK_TEMP_FOLDER_NAME,
-    outExtension: { '.js': '.cjs' }
+    outExtension: { '.js': '.cjs' },
+    ...buildOptions,
+    external: buildExternalDependencies(buildOptions)
   });
 };
 
+function buildExternalDependencies(buildOptions: BuildOptions = {}): string[] {
+  // explicitly add the `esbuild` as an external build dependency as recommended
+  // by esbuild to avoid bundling it in the output.
+  //
+  // ref: https://github.com/evanw/esbuild//issues/2698#issuecomment-1325160925
+  return ['esbuild', ...(buildOptions.external ?? [])];
```

Comment:
Since this is now part of a JS API, we don't want `esbuild` to consider `esbuild` in its output. Otherwise, we receive a similar warning as the `jsdom` warning. See comment above.

```
Compiling endpoints...
▲ [WARNING] "esbuild" should be marked as external for use with "require.resolve" [require-resolve-not-external]

    ../../node_modules/.pnpm/esbuild@0.19.11/node_modules/esbuild/lib/main.js:1825:36:
      1825 │   const libMainJS = require.resolve("esbuild");
```